### DISCUSSION TITLE
🚀 Change the way hooks are called to automatically handle the last_call variable

### DISCUSSION THREAD

**Discussion by swierczek** [ExpressionEngine/ExpressionEngine#4643] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
Some extension hooks that are called by multiple add-ons essentially require _all addon developers_ using that hook to remember to include [ee()->extensions->last_call](https://docs.expressionengine.com/latest/development/extensions.html#ee-extensions-last_call) or else later calls of that hook won't use the data changed by earlier addons. This leads to issues when an addon is missing the hook, because we need to manually fix other addons' code in order for our hook to work as expected.

**Describe the solution you'd like**
It would be ideal if the `$this->last_call` automatically overrides whatever the previous `$data` value is (if it exists for this hook) and therefore gets forwarded to the next hook as an updated ([$this->last_call](https://github.com/ExpressionEngine/ExpressionEngine/blob/7.dev/system/ee/legacy/libraries/Extensions.php#L232) for reference). That way, developers wouldn't ever need to remember to include the `last_call` conditional. I recognize that not all hooks use parameters, but I don't think those hooks need to currently worry about the `last_call` anyway.

I'm also unsure if there are any instances where this _wouldn't_ be the ideal behavior. Every instance I've worked with needs the updated data from a previous hook, and never the original data. If that's not the case, a potentially complex way around this would be to add another parameter to each hook to pass the original data along too, but that's only if any hook would ever want the OG data anyway.

**Teachability, Documentation, Adoption, Migration Strategy**
It would be a simpler implementation of hooks, as future developers/code wouldn't need to account for this at all.

- [x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by dunnmj42** [comment]
Agreed 100%. Not consistently passing most-recently-changed data to extension hooks as described here invalidates much of the utility of hook priority in use-cases like this.
